enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,609 @@
+- What would happen if they try to upgrade a cluster to a version where some component they're using is no longer in the payload, but that component isn't available via any OLM catalog?
+  - Answer: The upgrade should be blocked if the existing installed operator can't support the new OCP version. If the existing installed operator can support the new OCP version, we'd allow the upgrade, you just wouldn't get a new version of that operator.
+- What does migration look like for a component that's removed from the payload and becomes optional, or one that's OLM-based today and would fall under this new category?
+  - Answer: For a CVO-based component, this would require re-packaging their manifests to use OLM's supported registry+v1 or plain+v1 (TBD) bundle formats. For OLM-based operators, migration should be limited, but there are restrictions/requirements on which operators can be installed through this PO mechanism. Those requirements/restrictions are outlined in this proposal.
+- Are cluster admins allowed to pin package versions?
+  - Answer: Yes. This can be deferred to future phases though.
+- Does this phase need to enforce cluster invariants?
+  - Answer: No. This can be enforced in future phases, and phase 0 can document these limitations.
+- Does this phase need to be SNO compliant?
+  - Answer: No, but the rukpak webhooks will need to be deployed using an HA configuration by default.
+- How are phase 0 requirements enforced?
+  - Answer: Documentation can suffice for phase 0. Future phases may introduce cluster invariants (e.g. one GVK/package per cluster) that help codify these requirements.
+- What legacy OLM runtime changes are during phase 0?
+  - Answer: Unknown. There may be some issues with OLM's 0.x InstallPlan resource, but installing an OLM-based operator that was already installed through the PO mechanism should result in a failure. Future phases should aim to iron out this interaction, and determine whether a common layer can be used to proxy traffic to both OLM control planes.
+- Does the console need to be able to provide a UI experience for POs?
+  - Answer: Yes. [CONSOLE-3219](https://issues.redhat.com/browse/CONSOLE-3219) is tracking this effort.
+
+### Test Plan
+
+N/A
+
+### Graduation Criteria
+
+N/A
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+N/A
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+N/A
+
+### Version Skew Strategy
+
+N/A
+
+### Operational Aspects of API Extensions
+
+- Fails creation of Bundle resources when the webhook is not available.
+- Unsafe CRD schema changes are permitted when the webhook is not available.
+- CRD validating webhook may introduce scale issues with clusters that contain 10,000+ custom resources in the cluster. Requests to this webhook may reach the 60s timeout.
```

Comment:
This is partly why I think the CRD validator should not be included in phase 0. I don't think we've given it enough attention, and it could end up being a distraction.

Reply:
Nice catch. Agreed, let's cut this component out from the proposal, and we've already cut it out from our downstream rukpak manifests when we bootstrapped those repositories.

Reply:
Updated https://github.com/openshift/enhancements/pull/1170/commits/bc14b4537125d43153cc452cd686e02cf96630b2.